rdrf/rdrf/forms/widgets/widgets.py

```diff
@@ -328,6 +328,23 @@ class RadioSelect(widgets.RadioSelect):
     def usable_for_types():
         return {CommonDataElement.DATA_TYPE_RANGE}
 
+    def _get_item_size(self):
```

Comment:
How about?

```python
def _setup_column_layout(self):
        no_of_choices = len(self.choices)
        longest_choice_text = max(len(choice[1]) for choice in self.choices)
        has_long_text = longest_choice_text > 50
        has_short_texts_only = longest_choice_text <= 10

        cols_per_row = 3
        if has_long_text:
            cols_per_row = 4
        elif has_short_texts_only and no_of_choices <= 3:
            cols_per_row = 2

        return f"col-sm-{cols_per_row} col-md-{cols_per_row + 1} col-lg-{cols_per_row}"
```

Gets rid of a for loop and makes the intent a bit clearer?

Also, why is `col-md-` always `size+1`. A `size` of `4` will make `col-md-5` which doesn't work well with a grid of `12`.